app/scripts/modules/kubernetes/src/v2/loadBalancer/details/details.controller.ts

```diff
@@ -0,0 +1,88 @@
+import { IController, IScope, module } from 'angular';
+import { IModalService } from 'angular-ui-bootstrap';
+import { StateService } from '@uirouter/angularjs';
+
+import {
+  Application,
+  CONFIRMATION_MODAL_SERVICE,
+  ILoadBalancer,
+  LOAD_BALANCER_WRITE_SERVICE,
+} from '@spinnaker/core';
+
+import { IKubernetesLoadBalancer } from './IKubernetesLoadBalancer';
+
+interface ILoadBalancerFromStateParams {
+  accountId: string;
+  region: string;
+  name: string;
+}
+
+class KubernetesLoadBalancerDetailsController implements IController {
+  public state = { loading: true };
+  private loadBalancerFromParams: ILoadBalancerFromStateParams;
+  public loadBalancer: IKubernetesLoadBalancer;
+  public dispatchRules: string[] = [];
+
+  constructor(private $uibModal: IModalService,
+              private $state: StateService,
+              private $scope: IScope,
+              loadBalancer: ILoadBalancerFromStateParams,
+              private app: Application) {
+    'ngInject';
+    this.loadBalancerFromParams = loadBalancer;
+    this.app.getDataSource('loadBalancers')
+      .ready()
+      .then(() => this.extractLoadBalancer());
+  }
+
+  public deleteLoadBalancer(): void {
+    this.$uibModal.open({
+      templateUrl: require('../../manifest/delete/delete.html'),
+      controller: 'kubernetesV2ManifestDeleteCtrl',
+      controllerAs: 'ctrl',
+      resolve: {
+        coordinates: {
+          name: this.loadBalancerFromParams.name,
+          namespace: this.loadBalancerFromParams.region,
+          account: this.loadBalancerFromParams.accountId,
+        },
+        application: this.app
+      }
+    });
+  }
+
+  private extractLoadBalancer(): void {
+    const rawLoadBalancer = this.app.getDataSource('loadBalancers').data.find((test: ILoadBalancer) => {
+      return test.name === this.loadBalancerFromParams.name &&
+        test.account === this.loadBalancerFromParams.accountId;
+    });
+
+    if (rawLoadBalancer) {
+      this.state.loading = false;
+      this.loadBalancer = rawLoadBalancer as IKubernetesLoadBalancer;
```

Comment:
I would do a clone or at least a shallow copy here - you're mutating that load balancer for the entire application, which has ruined my day in the past. 

e.g.:
```
this.loadBalancer = {
  ...rawLoadBalancer,
  namespace: rawLoadBalancer.region,
  displayName: rawLoadBalancer.manifest.metadata.name,
  etc
};
```

Or if you want to make this transformation for all load balancers, I would do that in a `KubernetesV2LoadBalancerTransformer`.

Reply:
Ah sounds good - I don't think this will be used anywhere else, but once it is I'll pull it out into the transformer.